Cleanup & improve 3rd party component's dashboards

francoisferrand opened the pull request:
Issue: ZENKO-3799

<img width="1634" alt="Screenshot 2021-12-30 at 23 03 31" src="https://user-images.githubusercontent.com/3909027/147791476-44590ea6-09fb-4424-a225-2e74c49a3a85.png">
<img width="1495" alt="Screenshot 2022-01-03 at 14 49 46" src="https://user-images.githubusercontent.com/3909027/147938226-be30e536-48d9-4c9c-97f5-e3a17de5ede4.png">
<img width="1651" alt="Screenshot 2021-12-30 at 23 04 40" src="https://user-images.githubusercontent.com/3909027/147791474-7badebab-c14d-421e-9b98-6cade885ee5e.png">

francoisferrand committed:
Redis: add UP and total keys panels

Issue: ZENKO-3799

francoisferrand committed:
Zookeeper: add top row with UP panel

as well as other important indicators.

Issue: ZENKO-3799

francoisferrand committed:
kafka: fix dashboard title

Issue: ZENKO-3799

francoisferrand committed:
Kafka: fix filtering by instance (broker)

`broker` is a variable, and must thus be referenced as $broker, with no
brace (except in the few instance where it is used to repeat a panel).

In addition, since All and multi-select are supported, we must use `=~`
to select it in promql expressions.

Issue: ZENKO-3799

francoisferrand committed:
Kafka: use $__rate_interval instead of hardcoded value

Issue: ZENKO-3799

francoisferrand committed:
Kafka: move legend below graphs

Since we have "long" topics and pod names, displaying legend on the
side eats all the space.

Issue: ZENKO-3799

francoisferrand committed:
Kafka: partition health by topic

Add heatmap of under-replicated partitions per topic.

Issue: ZENKO-3799

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

miniscruff commented:
Small thing here but i would split hits and misses. Because they will likely have very different values it can be hard to see any significant changes in the smaller one.

francoisferrand committed:
Redis: fix on pod restart

- Use sum() to aggregate series related to different pods
- Do not fill connect graph when no data

Issue: ZENKO-3799

francoisferrand committed:
Redis: add legend for hit/miss

The legend also shows the 'max', to make it easier to see high number
of misses.

Issue: ZENKO-3799

lavigne958 commented on monitoring/dashboards/redis.json:
```diff
@@ -61,6 +61,66 @@
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(up{namespace=\"zenko\", job=\"artesca-data-base-cache-metrics\"})",
```
In this file the label `job` is an input value, we should replace the constant string with the variable `${job}`.

lavigne958 commented on monitoring/dashboards/redis.json:
```diff
@@ -148,6 +208,66 @@
+            "targets": [
+                {
+                "exemplar": true,
+                "expr": "sum(redis_db_keys{namespace=\"zenko\", job=\"artesca-data-base-cache-metrics\"})",
```
same here we should replace the constant for job with the variable `${job}`.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-3799 contains:


* `2.2.5`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.2.4`


Please check the `Fix Version/s` of ZENKO-3799, or the target
branch of this pull request.

francoisferrand committed:
MongoDb: only show current member's state

Issue: ZENKO-3799

bert-e commented:
# Conflict




There is a conflict between your branch `feature/ZENKO-3799` and the
destination branch `development/2.2`.

Please resolve the conflict on **the feature branch** (`feature/ZENKO-3799`).

```sh
git fetch && \
git checkout origin/feature/ZENKO-3799 && \
git merge origin/development/2.2
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:feature/ZENKO-3799
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand committed:
Mongodb: display member state by pod

Use it instead of name, as it is must more user-friendly.

Issue: ZENKO-3799

francoisferrand committed:
Mongodb: fix disk I/O graphs

Use container stats instead, aggregated by pod, and set proper unit.

Issue: ZENKO-3799

francoisferrand committed:
Mongodb: display graph for open/avail connections

Issue: ZENKO-3799

francoisferrand committed:
Fix name of 3rd party dashboard in zenkoversions.yaml

Name must match the name of the images in ISO.

Issue: ZENKO-3799

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-3799 contains:


* `2.2.6`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.2.7`


Please check the `Fix Version/s` of ZENKO-3799, or the target
branch of this pull request.

francoisferrand commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



Please check the status of the associated issue ZENKO-3799.

Goodbye francoisferrand.
